src/Domain.LinnApps/Reports/SpendsReportService.cs

```diff
@@ -0,0 +1,157 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Layouts;
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.ExternalServices;
+
+    using MoreLinq;
+
+    public class SpendsReportService : ISpendsReportService
+    {
+        private readonly IPurchaseLedgerPack purchaseLedgerPack;
+
+        private readonly IReportingHelper reportingHelper;
+
+        private readonly IQueryRepository<SupplierSpend> spendsRepository;
+
+        private readonly IRepository<VendorManager, string> vendorManagerRepository;
+
+
+        public SpendsReportService(
+            IQueryRepository<SupplierSpend> spendsRepository,
+            IRepository<VendorManager, string> vendorManagerRepository,
+            IPurchaseLedgerPack purchaseLedgerPack,
+            IReportingHelper reportingHelper)
+        {
+            this.spendsRepository = spendsRepository;
+            this.vendorManagerRepository = vendorManagerRepository;
+            this.purchaseLedgerPack = purchaseLedgerPack;
+            this.reportingHelper = reportingHelper;
+        }
+
+        public ResultsModel GetSpendBySupplierReport(string vm)
+        {
+            var currentLedgerPeriod = this.purchaseLedgerPack.GetLedgerPeriod();
+            var yearStartLedgerPeriod = this.purchaseLedgerPack.GetYearStartLedgerPeriod();
+            var previousYearStartLedgerPeriod = yearStartLedgerPeriod - 12;
+
+            // do I need to check the current ledger period as an upper limit? Not sure if we'd have stuff in the future
+            var supplierSpends = this.spendsRepository.FilterBy(
+                        x => x.LedgerPeriod >= previousYearStartLedgerPeriod && x.LedgerPeriod <= currentLedgerPeriod
+                             && (string.IsNullOrWhiteSpace(vm) || x.Supplier.VendorManager == vm))
+                    .ToList();
+
+            var vmName = "ALL";
+            if (!string.IsNullOrWhiteSpace(vm))
+            {
+                var vendorManager = this.vendorManagerRepository.FindById(vm);
+                vmName = $"{vm} - {vendorManager.Employee.FullName} ({vendorManager.UserNumber})";
+            }
+
+            var reportLayout = new SimpleGridLayout(
+                this.reportingHelper,
+                CalculationValueModelType.TextValue,
+                null,
+                $"Spend by supplier report for Vendor Manager: {vmName}. For this financial year and last, excludes factors & VAT.");
+
+            this.AddSupplierReportColumns(reportLayout);
+
+            var values = new List<CalculationValueModel>();
+
+            var distinctSupplierSpends = supplierSpends.DistinctBy(x => x.SupplierId).Select(
+                x => new SupplierSpend
```

Comment:
did you need morelinq for this .DistincyBy(..)? I've always just used GroupBy's to achieve a distinct by, but this seems like a good option too

Reply:
ohh yeah, I meant to mention this but put this pr up last night without thinking - apparently it comes free in .Net 6 and is basically the same, so I thought I'd just add MoreLinq and start using it if you're all okay with the dependency until we're in .Net 6. It feels quite intuitive to use imo
https://stackoverflow.com/questions/489258/linqs-distinct-on-a-particular-property/489421#489421